### DISCUSSION TITLE
How do you set factors in the mStat object?

### DISCUSSION THREAD

**Discussion by RachBioHaz** [cafferychen777/MicrobiomeStat#14] [Q&A] [answered]
Hi, I imported a phyloseq object using mStat_convert_phyloseq_to_data_obj. I lost the factors I had in the phyloseq object. How do I set up factors in the order I need in the new MicrobiomeStat object?
Thanks

**Comment by cafferychen777** [comment, marked as answer]
Hi @RachBioHaz ,

Thank you for reaching out. It seems that you want to retain or set up factors in a specific order in the `MicrobiomeStat` object after importing from a `phyloseq` object. Here's how you can achieve this:

1. After converting your `phyloseq` object to a `MicrobiomeStat` object using `mStat_convert_phyloseq_to_data_obj`, you'll get an object similar to `peerj32.obj`.
2. To set up factors in the order you need, you can use the following code:

```R
# Assuming 'sex' is the metadata column you want to set as a factor in a specific order
peerj32.obj$meta.dat$sex <- factor(as.factor(peerj32.obj$meta.dat$sex), levels = c("male", "female"))
```

3. To check the order and the data:

```R
print(peerj32.obj$meta.dat$sex)
```

The output will display the data in the specified factor order.

Remember to replace `'sex'` with the actual metadata column name you're interested in, and `c("male", "female")` with the desired order of your factor levels.

I hope this helps. Let me know if you have any further questions!

Best regards,
Chen YANG
<img width="861" alt="Screenshot 2023-10-07 at 14 21 52" src="https://github.com/cafferychen777/MicrobiomeStat/assets/82987086/e799934f-dacb-4653-be89-b3404894a9ab">

**Reply by RachBioHaz** [reply]
Thanks
